cmd/serf/command/agent/config.go

```diff
@@ -114,6 +115,10 @@ type Config struct {
 	QueryResponseSizeLimit int `mapstructure:"query_response_size_limit"`
 	QuerySizeLimit         int `mapstructure:"query_size_limit"`
 
+	// UserEventSizeLimit is maximum byte size limit of user event `name` + `payload` in bytes.
+	// It's optimal to be relatively small, since it's going to be gossiped through the cluster.
+	UserEventSizeLimit int `mapstructure:"user_event_size_limit"`
```

Comment:
I feel like we should document that it is the user's responsibility to ensure this is still lower than their network's minimum path MTU between any two nodes in the cluster otherwise things aren't going to work well (worst case performance decrease but I suspect we won't handle fragmentation well and cause more "failures"). I've not actually validated this is true but it is transmitted by UDP right?

Reply:
Docs say 

> Because user events are sent along the gossip layer, which uses UDP, the payload and entire message framing must fit within a single UDP packet. 

We could maybe repeat that here and point out that users should only increase that if they know their path MTU will be sufficient.

Reply:
```
I've not actually validated this is true but it is transmitted by UDP right?
```

Correct, it's UDP :+1: 

```
We could maybe repeat that here and point out that users should only increase that if they know their path MTU will be sufficient.
```

Yep :+1:

Reply:
Updated in 151d9a86e24c8a380f2c9bce17a00d9abd2a9a69